File: src/main/java/com/evieclient/modules/impl/hypixel/AutoGG.java
package com.evieclient.modules.impl.hypixel;

import com.evieclient.Evie;
import com.evieclient.events.bus.EventSubscriber;
import com.evieclient.events.impl.client.ChatReceivedEvent;
import com.evieclient.modules.Category;
import com.evieclient.modules.Module;
import kotlin.text.Regex;
import net.minecraft.client.Minecraft;

public class AutoGG extends Module

public AutoGG() {
        super("AutoGG", "Automatically says GG in the chat after a game on Hypixel.", Category.UTIL, true);
        }
    @EventSubscriber
    public void onGameEnd(ChatReceivedEvent event) {
        if (!Minecraft.getMinecraft().isSingleplayer()) {
            Evie.log("Found ChatReceivedEvent, checking Regex");

            if(event.getChatComponent().getUnformattedText().matches("^\\+1st Killer - ?\\[?\\w*\\+*\\]? \\w+ - \\d+(?: Kills?)?$")) {
                Minecraft.getMinecraft().thePlayer.sendChatMessage("gg");
            }
            if (event.getChatComponent().getUnformattedText().matches("^Match Details \\(click name to view\\)$")) {
                Minecraft.getMinecraft().thePlayer.sendChatMessage("gg");
            }
        }
    }
}

File: src/main/java/com/evieclient/modules/impl/hypixel/AutoGG.kt
package com.evieclient.modules.impl.hypixel

import com.evieclient.events.bus.EventSubscriber
import com.evieclient.events.impl.client.ChatReceivedEvent
import com.evieclient.modules.Category
import com.evieclient.modules.Module
import net.minecraft.client.Minecraft

class AutoGG :
    Module("AutoGG", "Automatically says GG in the chat after a game on Hypixel.", Category.UTIL, true) {
    @EventSubscriber
    fun onGameEnd(event: ChatReceivedEvent?) {
        if (!Minecraft.getMinecraft().isSingleplayer) {
            if (event?.toString()?.matches(Regex("^ +1st Killer - ?\\[?\\w*\\+*\\]? \\w+ - \\d+(?: Kills?)?$")) == true) {
                Minecraft.getMinecraft().thePlayer.sendChatMessage("gg")
            }
            if (event?.toString()?.matches(Regex("^Match Details \\(click name to view\\)$")) == true) {
                Minecraft.getMinecraft().thePlayer.sendChatMessage("gg")
            }
        }
    }
}